specifyweb/specify/views.py

```diff
@@ -456,6 +456,16 @@ def record_merge_fx(model_name: str, old_model_ids: List[int], new_model_id: int
         except (ValueError):
             continue
 
+        # Handle case of updating a large amount of agent ids in the audit logs.
+        # Fix by optimizing the query by consolidating it here
+        if model_name.lower() == 'agent' and table_name.lower() == 'spauditlog':
```

Comment:
how hard would be to generalize this optimization for other common cases? (Determination -> Taxon, Collecting Event -> Locality, ...)
because it doesn't look like you have much agent-specific code in here

Reply:
Good suggestion.  I did some generalization, by adding `MERGING_OPTIMIZATION_TABLES` and `MERGING_OPTIMIZATION_FIELDS`.  Might do more work on this in the future.